R/aaa.R

```diff
@@ -8,8 +8,20 @@ new_function <- function(args = NULL,
   as.function.default(c(args, body) %||% list(NULL), env)
 }
 
+
+`append<-` <- function(x, after, value) {
+  if (missing(after))
+    c(x, value)
+  else
+    append(x, value, after = after)
+}
+
 `append1<-` <- function (x, value) {
```

Comment:
Should this be `append_list<-`?

Reply:
I don't think so, because it works with atomic vectors too. The constraint is that it only ever appends 1 element, and a guardrail is there to avoid any implicit type casting.

```r
> x <- character()
> append1(x) <- "foo"
> append1(x) <- c("foo", "bar")
Error in `x[[length(x) + 1L]] <- value`:
! more elements supplied than there are to replace
Run `rlang::last_trace()` to see where the error occurred.
```

Reply:
Ooooh, that's what the `1` means.

Maybe it would be worth rewriting `append<-` to use `[<-` so it's more obvious what the distinction is?

Reply:
`c()` does a lot more work than `[<-`, and it'd be tricky to match the current feature set with just `[<-`. (FWIW, this pair of `append<-` and `append1<-` is pretty seasoned).